src/staticdata.c

```diff
@@ -815,22 +823,21 @@ static void jl_write_values(jl_serializer_state *s)
                         write_pointer(s->const_data);
                 }
                 else {
-                    int isbitsunion = jl_array_isbitsunion(ar);
-                    if (ar->elsize == 1 && !isbitsunion)
-                        tot += 1;
                     ios_write(s->const_data, (char*)jl_array_data(ar), tot);
-                    if (isbitsunion)
-                        ios_write(s->const_data, jl_array_typetagdata(ar), alen);
```

Comment:
There may be a gap between jl_array_data and jl_array_typetagdata, so tot += alen is not the same as this